src/core/components/files-regular/add-readable-stream.js

```diff
@@ -1,55 +1,45 @@
 'use strict'
 
-const pull = require('pull-stream')
-const pushable = require('pull-pushable')
-const Duplex = require('readable-stream').Duplex
+const {
+  Duplex,
+  PassThrough
+} = require('readable-stream')
 
 class AddHelper extends Duplex {
-  constructor (pullStream, push, options) {
-    super(Object.assign({ objectMode: true }, options))
-    this._pullStream = pullStream
-    this._pushable = push
-    this._waitingPullFlush = []
+  constructor (stream, options) {
+    super(options)
+    this._stream = stream
   }
 
   _read () {
-    this._pullStream(null, (end, data) => {
-      while (this._waitingPullFlush.length) {
-        const cb = this._waitingPullFlush.shift()
-        cb()
-      }
-      if (end) {
-        if (end instanceof Error) {
-          this.emit('error', end)
-        } else {
-          this.push(null)
-        }
-      } else {
-        this.push(data)
-      }
-    })
+
   }
 
   _write (chunk, encoding, callback) {
-    this._waitingPullFlush.push(callback)
-    this._pushable.push(chunk)
+    this._stream.write(chunk, encoding, callback)
   }
 }
 
 module.exports = function (self) {
   return (options) => {
-    options = options || {}
-
-    const p = pushable()
-    const s = pull(
-      p,
-      self.addPullStream(options)
-    )
+    const stream = new PassThrough({
+      objectMode: true
+    })
 
-    const retStream = new AddHelper(s, p)
+    const output = new AddHelper(stream, {
```

Comment:
Can [`it-to-stream`](https://www.npmjs.com/package/it-to-stream) or [`stream-to-it`](https://www.npmjs.com/package/stream-to-it) help here to avoid buffering and respect backpressure?

Reply:
Good shout, I'll pull those in.